app/router/student.py

```diff
@@ -125,3 +130,79 @@ async def verify_student(request: Request, student_id: str):
             return HTTPException(status_code=404, detail="Student ID does not exist!")
         return True
     return HTTPException(status_code=response.status_code, detail=response.errors)
+
+
+def get_jnv_code(data):
+    response = requests.get(
+        school_db_url, params={"region": data.region, "name": data.school_name}
+    )
+    if response.status_code == 200:
+        if len(response.json()) == 0:
+            return HTTPException(
+                status_code=404, detail="JNV or region does not exist!"
+            )
+        return response.json()
+    return HTTPException(status_code=response.status_code, detail=response.errors)
+
+
+def generate_three_digit_code(code):
+    for _ in range(3):
+        code += random.randint(0, 9)
+    return code
+
+
+def build_data_object(params):
+    data = {}
+    for field in params.keys():
+        if (
+            field in form_details.fields
+            and params[field]
+            and (params[field] != "" or params[field] is not None)
+        ):
+            data[field] = str(params[field])
+        else:
+            return HTTPException(
+                status_code=400, detail="Parameter {} is invalid!".format(field)
+            )
+    data["created_at"] = datetime.now(pytz.timezone("Asia/Kolkata"))
+    return data
+
+
+@router.post("/")
+async def create_student(request: Request):
+    """ """
+    params = await request.json()
+    data = build_data_object(params)
+    print(data)
+
+    # TODO: Think about this logic. How many/what parameters are required to check if student already exists?
+    does_student_already_exist = requests.get(student_db_url, params=params)
+    if does_student_already_exist.status_code == 200:
+        student_data = does_student_already_exist.json()
+        if len(student_data) == 0:
+            student_id = ""
+
+            if data.group == "EnableStudents":
+                student_id += form_details.grade_code[data.grade]
+                student_id += get_jnv_code(data)  # test
+                student_id += generate_three_digit_code("")
+                does_generated_id_exist = verify_student(student_id=student_id)
+                if does_generated_id_exist:
+                    return student_id
+
+                is_student_added, number_of_loops = False, 0
+                if not is_student_added or number_of_loops < 1000:
+                    if not does_generated_id_exist:
+                        is_student_added = True
+                        # add ID to db
+                        return student_id
+                    else:
+                        student_id[:-3] += generate_three_digit_code("")
```

Comment:
we could just have the function as `generate_three_digit_code()` without any input na